store/sqlstore/channel_store.go

```diff
@@ -2154,6 +2154,27 @@ func (s SqlChannelStore) SearchInTeam(teamId string, term string, includeDeleted
 	})
 }
 
+func (s SqlChannelStore) SearchArchivedInTeam(teamId string, term string) (*model.ChannelList, *model.AppError) {
+	deleteFilter := "AND c.DeleteAt != 0"
+
+	return s.performSearch(`
+		SELECT
+			Channels.*
+		FROM
+			Channels
+		JOIN
+			PublicChannels c ON (c.Id = Channels.Id)
```

Comment:
Should this also search the user's deleted private channels as well?

Reply:
Yeah, I think you're right. I changed the function to also search private channels the user was a member of. I couln't get it to work with a single query, so I had to write two queries and combine the results. Let me know if you have some improvements there!